packages/grafana-llm-app/pkg/mcp/live_server.go

```diff
@@ -94,14 +124,42 @@ type liveSession struct {
 	// sender is the StreamSender for the Grafana Live session. It is used to send
 	// JSON-RPC responses back to the client.
 	sender *backend.StreamSender
+
+	// accessToken is the access token for the Grafana Live session.
+	accessToken string
 }
 
 // HandleStream handles a new Grafana Live session.
 func (s *GrafanaLiveServer) HandleStream(ctx context.Context, req *backend.RunStreamRequest, sender *backend.StreamSender) error {
+	var tokenExchangeResponse *authn.TokenExchangeResponse
+	// Only do the token exchange if we are in Grafana Cloud.
+	if s.EnableGrafanaManagedLLM {
+		var tokenExchangeClient *authn.TokenExchangeClient
+		var err error
+		// We need to use the token exchange API to create a token for the Grafana Live session.
+		tokenExchangeClient, err = authn.NewTokenExchangeClient(authn.TokenExchangeConfig{
```

Comment:
I could be missing something obvious but generally speaking, you'll want to instantiate the client once only somewhere as high up in the chain as possible so that you can benefit from the caching strategy.